eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -274,16 +284,25 @@ object MultiPartHandler {
    * @param routes              routes that must be blinded and provided in the invoice.
    * @param router              router actor.
    * @param paymentPreimage_opt payment preimage.
+   * @param pathId_opt          path id for the payment paths. If None, a different one will be randomly generated for
+   *                            each path and the mapping blinding point -> path id will be store in the database.
+   * @param storeInDb           whether to store the incoming payment in the database. Bolt12 invoices generated as a
+   *                            response to an invoice request should not be stored as it would be a DoS vector.
    */
   case class ReceiveOfferPayment(nodeKey: PrivateKey,
                                  invoiceRequest: InvoiceRequest,
                                  routes: Seq[ReceivingRoute],
                                  router: ActorRef,
                                  paymentPreimage_opt: Option[ByteVector32] = None,
+                                 pathId_opt: Option[ByteVector] = None,
+                                 additionalTlvs: Set[InvoiceTlv] = Set.empty,
+                                 customTlvs: Set[GenericTlv] = Set.empty,
+                                 storeInDb: Boolean = true,
                                  paymentType: String = PaymentType.Blinded) extends ReceivePayment {
```

Comment:
We should probably remove that argument and set it ourselves inside this function, especially if we use a new payment type for the DB-less offer invoices.

Reply:
Removed